### ISSUE TITLE
Problems running the server locally.

### ISSUE THREAD

**Issue by Losnen** [education/classroom#1243]
I'm trying to run the server locally for contribute but I have some problems with elasticsearch after the authorization with github.

got this in the terminal

`17:44:53 web.1           | 2018-01-26 17:44:53 +0000: Rack app error handling request { GET /auth/github/callback }
17:44:53 web.1           | #<Elasticsearch::Transport::Transport::Errors::BadRequest: [400] {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"No handler for type [string] declared on field [group_title]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [group_assignment_repo]: No handler for type [string] declared on field [group_title]","caused_by":{"type":"mapper_parsing_exception","reason":"No handler for type [string] declared on field [group_title]"}},"status":400}>
`

and got this on the browser

`Puma caught this error: [400] {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"No handler for type [string] declared on field [group_title]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [group_assignment_repo]: No handler for type [string] declared on field [group_title]","caused_by":{"type":"mapper_parsing_exception","reason":"No handler for type [string] declared on field [group_title]"}},"status":400} (Elasticsearch::Transport::Transport::Errors::BadRequest)`

Can you help me please?

**Comment by Kadaaran** [comment]
Hi @Losnen did you have any trouble while running `./script/setup` ?

**Comment by Losnen** [comment]
`$ ./script/setup 
==> Bootstrapping Homebrew OSX environment
==> Installing gem dependencies…
Using rake 12.0.0
Using concurrent-ruby 1.0.5
Using i18n 0.8.6
Using minitest 5.10.3
Using thread_safe 0.3.6
Using tzinfo 1.2.3
Using activesupport 5.1.3
Using builder 3.2.3
Using erubi 1.6.1
Using mini_portile2 2.2.0
Using nokogiri 1.8.0
Using rails-dom-testing 2.0.3
Using loofah 2.0.3
Using rails-html-sanitizer 1.0.3
Using actionview 5.1.3
Using rack 2.0.3
Using rack-test 0.6.3
Using actionpack 5.1.3
Using nio4r 2.1.0
Using websocket-extensions 0.1.2
Using websocket-driver 0.6.5
Using actioncable 5.1.3
Using globalid 0.4.0
Using activejob 5.1.3
Using mime-types-data 3.2016.0521
Using mime-types 3.1
Using mail 2.6.6
Using actionmailer 5.1.3
Using activemodel 5.1.3
Using arel 8.0.0
Using activerecord 5.1.3
Using public_suffix 3.0.0
Using addressable 2.5.2
Using ast 2.3.0
Using execjs 2.7.0
Using autoprefixer-rails 7.1.3
Using awesome_print 1.8.0
Using bindex 0.5.0
Using msgpack 1.1.0
Using bootsnap 1.1.2
Using uniform_notifier 1.10.0
Using bullet 5.6.1
Using bundler 1.16.1
Using byebug 9.1.0
Using multi_json 1.12.1
Using elasticsearch-api 5.0.4
Using multipart-post 2.0.0
Using faraday 0.12.2
Using elasticsearch-transport 5.0.4
Using elasticsearch 5.0.4
Using elasticsearch-dsl 0.1.5
Using chewy 0.10.1
Using coderay 1.1.2
Using coffee-script-source 1.12.2
Using coffee-script 2.4.1
Using method_source 0.8.2
Using thor 0.19.4
Using railties 5.1.3
Using coffee-rails 4.2.2
Using color 1.8
Using concurrent-ruby-ext 1.0.5
Using connection_pool 2.2.1
Using safe_yaml 1.0.4
Using crack 0.4.3
Using dalli 2.7.6
Using database_cleaner 1.6.1
Using diff-lcs 1.3
Using docile 1.1.5
Using dotenv 2.2.1
Using dotenv-rails 2.2.1
Using erubis 2.7.0
Using ffi 1.9.18
Using ethon 0.10.1
Using factory_bot 4.8.2
Using factory_bot_rails 4.8.2
Using failbot 2.0.1
Using sprockets 3.7.1
Using sprockets-rails 3.2.1
Using rails 5.1.3
Using failbot_rails 0.5.0
Using faker 1.8.4
Using faraday-http-cache 2.0.0
Using flipper 0.10.2
Using redis 3.3.3
Using flipper-redis 0.10.2
Using rack-protection 2.0.0
Using flipper-ui 0.10.2
Using foreman 0.84.0
Using formatador 0.2.5
Using geo_pattern 1.4.0
Using rb-fsevent 0.10.2
Using rb-inotify 0.9.10
Using ruby_dep 1.5.0
Using listen 3.1.5
Using lumberjack 1.0.12
Using nenv 0.3.0
Using shellany 0.0.1
Using notiffany 0.1.1
Using slop 3.6.0
Using pry 0.10.4
Using guard 2.14.1
Using guard-compat 1.2.1
Using rspec-support 3.6.0
Using rspec-core 3.6.0
Using rspec-expectations 3.6.0
Using rspec-mocks 3.6.0
Using rspec 3.6.0
Using guard-rspec 4.7.3
Using hashdiff 0.3.6
Using hashie 3.5.6
Using jquery-datetimepicker-rails 2.4.1.0
Using turbolinks 3.0.0 from https://github.com/turbolinks/turbolinks-classic.git (at 37a7c29@37a7c29)
Using jquery-turbolinks 2.1.0
Using json 2.1.0
Using jwt 1.5.6
Using kaminari-core 1.0.1
Using kaminari-actionview 1.0.1
Using kaminari-activerecord 1.0.1
Using kaminari 1.0.1
Using knapsack 1.14.1
Using local_time 2.0.0
Using multi_xml 0.6.0
Using oauth2 1.4.0
Using octicons 2.1.0
Using octicons_helper 2.1.0
Using sawyer 0.8.1
Using octokit 4.7.0
Using omniauth 1.6.1
Using omniauth-oauth2 1.4.0
Using omniauth-github 1.3.0
Using parallel 1.12.0
Using parser 2.4.0.0
Using peek 1.0.1
Using peek-dalli 1.1.3 from https://github.com/peek/peek-dalli.git (at 0a68e1f@0a68e1f)
Using peek-gc 0.0.2
Using peek-git 1.0.2
Using peek-performance_bar 1.2.0
Using pg 0.21.0
Using peek-pg 1.2.0 from https://github.com/mkcode/peek-pg.git (at 9bbe212@9bbe212)
Using sidekiq 5.0.4
Using peek-sidekiq 1.0.0 from https://github.com/Soliah/peek-sidekiq.git (at 261c857@261c857)
Using powerpack 0.1.1
Using pry-byebug 3.5.0
Using pry-rails 0.3.6
Using puma 3.10.0
Using rack-canonical-host 0.2.3
Using rack-timeout 0.4.2
Using rails-controller-testing 1.0.2
Using rails-i18n 5.0.4
Using rainbow 2.2.2
Using redis-namespace 1.5.3
Using rspec-rails 3.6.1
Using ruby-progressbar 1.8.1
Using unicode-display_width 1.3.0
Using rubocop 0.49.1
Using sass 3.4.25
Using tilt 2.0.8
Using sass-rails 5.0.6
Using scss_lint 0.54.0
Using simplecov-html 0.10.2
Using simplecov 0.15.0
Using spring 2.0.2
Using spring-watcher-listen 2.0.1
Using terminal-notifier-guard 1.7.0
Using timecop 0.9.1
Using typhoeus 1.3.0
Using uglifier 3.2.0
Using vcr 3.0.3
Using web-console 3.5.1
Using webmock 3.0.1
Bundle complete! 70 Gemfile dependencies, 170 gems now installed.
Gems in the group production were not installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
==> Installing javascript dependencies…
yarn install v1.3.2
[1/4] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.22s.
===> Setting up DB...
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Dropped database 'classroom_development'
Dropped database 'classroom_test'
Created database 'classroom_development'
Created database 'classroom_test'
-- enable_extension("plpgsql")
   -> 0.0397s
-- create_table("assignment_invitations", {:id=>:serial, :force=>:cascade})
   -> 0.0234s
-- create_table("assignment_repos", {:id=>:serial, :force=>:cascade})
   -> 0.0207s
-- create_table("assignments", {:id=>:serial, :force=>:cascade})
   -> 0.0212s
-- create_table("deadlines", {:id=>:serial, :force=>:cascade})
   -> 0.0063s
-- create_table("group_assignment_invitations", {:id=>:serial, :force=>:cascade})
   -> 0.0150s
-- create_table("group_assignment_repos", {:id=>:serial, :force=>:cascade})
   -> 0.0127s
-- create_table("group_assignments", {:id=>:serial, :force=>:cascade})
   -> 0.0190s
-- create_table("groupings", {:id=>:serial, :force=>:cascade})
   -> 0.0057s
-- create_table("groups", {:id=>:serial, :force=>:cascade})
   -> 0.0101s
-- create_table("groups_repo_accesses", {:id=>false, :force=>:cascade})
   -> 0.0094s
-- create_table("organizations", {:id=>:serial, :force=>:cascade})
   -> 0.0232s
-- create_table("organizations_users", {:id=>false, :force=>:cascade})
   -> 0.0058s
-- create_table("repo_accesses", {:id=>:serial, :force=>:cascade})
   -> 0.0102s
-- create_table("roster_entries", {:force=>:cascade})
   -> 0.0113s
-- create_table("rosters", {:force=>:cascade})
   -> 0.0076s
-- create_table("users", {:id=>:serial, :force=>:cascade})
   -> 0.0152s
-- enable_extension("plpgsql")
   -> 0.0375s
-- create_table("assignment_invitations", {:id=>:serial, :force=>:cascade})
   -> 0.0244s
-- create_table("assignment_repos", {:id=>:serial, :force=>:cascade})
   -> 0.0204s
-- create_table("assignments", {:id=>:serial, :force=>:cascade})
   -> 0.0122s
-- create_table("deadlines", {:id=>:serial, :force=>:cascade})
   -> 0.0116s
-- create_table("group_assignment_invitations", {:id=>:serial, :force=>:cascade})
   -> 0.0227s
-- create_table("group_assignment_repos", {:id=>:serial, :force=>:cascade})
   -> 0.0174s
-- create_table("group_assignments", {:id=>:serial, :force=>:cascade})
   -> 0.0197s
-- create_table("groupings", {:id=>:serial, :force=>:cascade})
   -> 0.0098s
-- create_table("groups", {:id=>:serial, :force=>:cascade})
   -> 0.0104s
-- create_table("groups_repo_accesses", {:id=>false, :force=>:cascade})
   -> 0.0073s
-- create_table("organizations", {:id=>:serial, :force=>:cascade})
   -> 0.0184s
-- create_table("organizations_users", {:id=>false, :force=>:cascade})
   -> 0.0096s
-- create_table("repo_accesses", {:id=>:serial, :force=>:cascade})
   -> 0.0121s
-- create_table("roster_entries", {:force=>:cascade})
   -> 0.0110s
-- create_table("rosters", {:force=>:cascade})
   -> 0.0032s
-- create_table("users", {:id=>:serial, :force=>:cascade})
   -> 0.0163s
==> Your app is almost ready! Please make sure you fill out your .env file.
`
@Kadaaran it runs perfectly I think.

**Comment by Kadaaran** [comment]
@Losnen Can you give me your `elasticsearch` version and your `java` version as well ? 
I know I had trouble while setting up the app (we never know it might be the same for you)

**Comment by Losnen** [comment]
@Kadaaran 

My `elasticsearch` version:

```
$ elasticsearch --version
Version: 6.1.2, Build: 5b1fea5/2018-01-10T02:35:59.208Z, JVM: 1.8.0_162
```

My `java `version:

```
$ java --version
java 9.0.4
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)
```

**Comment by Kadaaran** [comment]
@Losnen I don't know if it will work but can you update your java version to `Java 8` , make sure `Java 9` is uninstalled and see if it works ?

**Comment by Kadaaran** [comment]
Merged #1251
